### PR TITLE
feat: wezterm integration

### PR DESCRIPTION
**Pull request by mrossinek**
This adds support for font size adaptation in the wezterm terminal.

Features:
- absolute valued font size config (by using a plain number setting, e.g. `18` or `"18"`)
- relative font size incrementing (by using a setting which starts with `+`, e.g. `"+14"`)
- tab bar hiding in Zen mode (can be disabled by adapting the wezterm configuration function)

References:
- https://github.com/wez/wezterm/discussions/2550
- https://www.reddit.com/r/neovim/comments/xn1q75/comment/iprdrpr

### CODE REVIEW

**Comment by mikeduminy** [comment]
This looks great, @folke how do you feel about it?

File: lua/zen-mode/plugins.lua [hunk 1]
```diff
@@ -47,6 +47,16 @@ function M.kitty(state, disable, opts)
   vim.cmd([[redraw]])
 end
 
+function M.wezterm(state, disable, opts)
+  local stdout = vim.loop.new_tty(1, false)
+  if disable then
+    stdout:write(('\x1b]1337;SetUserVar=%s=%s\b'):format('ZEN_MODE', vim.fn.system({ 'base64' }, tostring(opts.font))))
```
Comment (megalithic) [inline lua/zen-mode/plugins.lua:53]: can this be modified to handle being in tmux too?

**Review by megalithic — CHANGES_REQUESTED** [review]

File: lua/zen-mode/plugins.lua (hunk 1 above)
Comment (mrossinek) [reply lua/zen-mode/plugins.lua:53]: I am not sure what you mean? The tmux integration is done separately in another file

**Comment by folke** [comment]
Thank you for this PR. 

Can you remove the wezterm integration comment from the config and config section in the readme?

And add a new section below usage:

## 🧩 Plugins

### Wezterm

Then add the specific setup instructions here

**Comment by mrossinek** [comment]
> Thank you for this PR.
> 
> Can you remove the wezterm integration comment from the config and config section in the readme?

Will do this when I get home tonight :+1:

File: lua/zen-mode/plugins.lua (hunk 1 above)
Comment (megalithic) [reply lua/zen-mode/plugins.lua:70]: I was talking about the situations where you are running tmux within wezterm. I believe there are additional bits of ansi you need to add. If you don’t care to support that, that’s fine, it’s be nice though! 😁🙏🏻

Comment (mrossinek) [reply lua/zen-mode/plugins.lua:70]: I do not currently have the time to dig into this, sorry. I suggest merging this as is and add this in the future (if someone needs it and knows how to/has the time to make this work).

**Comment by mrossinek** [comment]
Requested changes are done (commenting here because I cannot request a review as it seems)

**Review by folke — CHANGES_REQUESTED** [review]

**Comment by winter-again** [comment]
Hi @folke . I'd like to adapt [this code](https://github.com/folke/zen-mode.nvim/blob/cb73b8bd0ef9d765b942db09dc762c603a89ae44/lua/zen-mode/plugins.lua#L68-L73) from this PR in a Neovim plugin I'm working on. I see that the project is licensed under Apache 2.0., but since I'm only referencing/modifying a handful of lines, I wasn't sure what amount of attribution to give. I was already planning to at least have an inline reference to the zen-mode plugin in my source code, but I'm new to working with licenses and wasn't sure if something more is necessary. Would you be okay with just an inline reference? Thanks.

**Comment by folke** [comment]
feel free to pick whatever code you need :)

**Pull request merged**